{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "70cab5b6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8d7fdcea",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "from dotenv import load_dotenv\n",
    "from pymongo import MongoClient\n",
    "from pymongo.errors import OperationFailure\n",
    "\n",
    "load_dotenv(\"../../.env.localhost\")\n",
    "\n",
    "admin_client = MongoClient(\n",
    "    host=os.getenv(\"MONGO_HOST\"),\n",
    "    username=os.getenv(\"MONGO_USERNAME\"),\n",
    "    password=os.getenv(\"MONGO_PASSWORD\"),\n",
    "    directConnection=True, # connect to host as a standalone, rather than to entire replicaset\n",
    ")\n",
    "reader_client = MongoClient(\n",
    "    host=os.getenv(\"MONGO_HOST\"),\n",
    "    username=os.getenv(\"MONGO_READONLY_USERNAME\"),\n",
    "    password=os.getenv(\"MONGO_READONLY_PASSWORD\"),\n",
    "    directConnection=True, # connect to host as a standalone, rather than to entire replicaset\n",
    ")\n",
    "\n",
    "mdb_src = reader_client[os.getenv(\"MONGO_DBNAME\")]\n",
    "mdb_tgt = admin_client[os.getenv(\"MONGO_DBNAME\") + \"-next\"]\n",
    "\n",
    "\n",
    "try:\n",
    "    mdb_src.study_set.insert_one({\"_id\": \"foobar\"})\n",
    "    print(\"NOT OK\")\n",
    "    mdb_src.study_set.delete_one({\"_id\": \"foobar\"})\n",
    "    print(\"state restored\")\n",
    "    raise Exception(\"can write to `mdb_src`!\")\n",
    "except OperationFailure:\n",
    "    pass"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ca3d913-d157-44cd-ad1f-4393d4ab64e4",
   "metadata": {},
   "source": [
    "No output from above? Good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ceb92334",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "\"\"\"Utils\"\"\"\n",
    "\n",
    "from copy import deepcopy\n",
    "import pkg_resources\n",
    "from pprint import pprint\n",
    "import time\n",
    "\n",
    "import fastjsonschema\n",
    "from toolz import dissoc\n",
    "\n",
    "def strip_oid(doc):\n",
    "    return dissoc(doc, \"_id\")\n",
    "\n",
    "def nmdc_schema_collection_names() -> set:\n",
    "    return {\n",
    "        k for k, v in get_nmdc_jsonschema_dict()[\"$defs\"][\"Database\"][\"properties\"].items()\n",
    "        if v.get(\"items\",{}).get(\"$ref\")\n",
    "    }\n",
    "\n",
    "def present_src_collections(mdb) -> list:\n",
    "    return sorted(\n",
    "        n for n in (nmdc_schema_collection_names() & set(mdb_src.list_collection_names()))\n",
    "        if mdb_src[n].estimated_document_count()\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "27a094a8-296e-4320-9728-0c8bca4cf5a2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7.4.11\n"
     ]
    }
   ],
   "source": [
    "\"\"\"Monkey-patch schema in light of expected metadata/schema modifications.\"\"\"\n",
    "\n",
    "from nmdc_runtime.util import get_nmdc_jsonschema_dict\n",
    "\n",
    "def without_id_patterns(nmdc_jsonschema):\n",
    "    rv = deepcopy(nmdc_jsonschema)\n",
    "    for cls_, spec in rv[\"$defs\"].items():\n",
    "        if \"properties\" in spec:\n",
    "            if \"id\" in spec[\"properties\"]:\n",
    "                spec[\"properties\"][\"id\"].pop(\"pattern\", None)\n",
    "    return rv\n",
    "\n",
    "\n",
    "# SCHEMA_DEFS_PATCHES = {\n",
    "#     \"Entity1\": {\n",
    "#         \"attribute1\": {\n",
    "#                \"description\": \"Lorem ipsum\",\n",
    "#                \"type\": \"string\"\n",
    "#         },\n",
    "#         \"attribute2\": {\n",
    "#                \"items\": {\n",
    "#                   \"$ref\": \"#/$defs/Entity2\"\n",
    "#                },\n",
    "#                \"type\": \"array\"\n",
    "#         }\n",
    "#     },\n",
    "# }\n",
    "\n",
    "# def patch_schema_defs(nmdc_jsonschema, patches=SCHEMA_DEFS_PATCHES):\n",
    "#     rv = deepcopy(nmdc_jsonschema)\n",
    "#     defs = rv[\"$defs\"]\n",
    "#     for cls_, spec in patches.items():\n",
    "#         if cls_ not in defs:\n",
    "#             raise Exception(f\"Class not present, so cannot patch: {cls_}\")\n",
    "#         elif \"properties\" not in defs[cls_]:\n",
    "#             defs[cls_][\"properties\"] = spec\n",
    "#         else:\n",
    "#             for k, v in spec.items():\n",
    "#                 defs[cls_][\"properties\"][k] = v\n",
    "#     return rv\n",
    "\n",
    "print(pkg_resources.get_distribution(\"nmdc_schema\").version)\n",
    "\n",
    "nmdc_jsonschema_validate = fastjsonschema.compile(\n",
    "    without_id_patterns(get_nmdc_jsonschema_dict())\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3949bb49-eacd-42da-a588-d8441a010445",
   "metadata": {},
   "source": [
    "Reference for anticipated migration logic/config: https://github.com/microbiomedata/nmdc-schema/blob/v7.4.10/nmdc_schema/migrate_3_2_to_7.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "a0fbba94-b5bb-44c6-9ccc-28c2d67b58d4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "from inspect import signature\n",
    "import re\n",
    "\n",
    "from toolz import assoc, assoc_in, dissoc\n",
    "\n",
    "\n",
    "id_pattern = re.compile(r\"(PO|ENVO):\\d+\")\n",
    "curie_pattern = re.compile(r\"[A-Za-z]\\w+:\\w+\")\n",
    "\n",
    "\n",
    "def ensure_term_id(v):\n",
    "    if not (\"term\" in v or \"has_raw_value\" in v):\n",
    "        raise Exception(f\"needs `term` or `has_raw_value`\")\n",
    "    if not (m := re.search(id_pattern, v[\"has_raw_value\"])):\n",
    "        raise Exception(f'{v[\"has_raw_value\"]} does not match a known ID pattern')\n",
    "    \n",
    "    return assoc(v, \"term\", {\"id\": v[\"has_raw_value\"]})\n",
    "\n",
    "\n",
    "def uppercase_curie_prefixes(v):\n",
    "    if not isinstance(v, list):\n",
    "        raise Exception(\"list expected\")\n",
    "    if not all(\":\" in elt for elt in v):\n",
    "        raise Exception(\"CURIEs expected\")\n",
    "    \n",
    "    rv = []\n",
    "    for elt in v:\n",
    "        prefix, localpart = elt.split(\":\", maxsplit=1)\n",
    "        rv.append(prefix.upper() + \":\" + localpart)\n",
    "    return rv\n",
    "\n",
    "\n",
    "def lowercase_curie_prefixes(v):\n",
    "    if not isinstance(v, list):\n",
    "        raise Exception(\"list expected\")\n",
    "    if not all(\":\" in elt for elt in v):\n",
    "        raise Exception(\"CURIEs expected\")\n",
    "    \n",
    "    rv = []\n",
    "    for elt in v:\n",
    "        prefix, localpart = elt.split(\":\", maxsplit=1)\n",
    "        rv.append(prefix.lower() + \":\" + localpart)\n",
    "    return rv\n",
    "\n",
    "\n",
    "def ensure_curies(v):\n",
    "    \"\"\"\n",
    "    >>> ensure_curies(['HTTPS://identifiers.org/gold:Gs0144557'])\n",
    "    ['gold:Gs0144557']\n",
    "    \"\"\"\n",
    "    if not isinstance(v, list):\n",
    "        raise Exception(\"list expected\")\n",
    "    if not all(\":\" in elt for elt in v):\n",
    "        raise Exception(\"CURIE-ables expected\")\n",
    "\n",
    "    return [re.search(curie_pattern, elt).group(0) for elt in v]\n",
    "\n",
    "\n",
    "def ensure_depth_via_depth2(v, d):\n",
    "    if \"depth\" not in d:\n",
    "        raise Exception(\"no `depth` field\")\n",
    "    \n",
    "    depth = d[\"depth\"]\n",
    "    return {\"depth\": depth, \"depth2\": None}\n",
    "\n",
    "\n",
    "def remove_type_field(v):\n",
    "    return dissoc(v, \"type\")\n",
    "\n",
    "\n",
    "def ensure_number_in_meters(v):\n",
    "    if isinstance(v, dict):\n",
    "        assert \"has_unit\" in v, \"no units\"\n",
    "        assert v[\"has_unit\"] in (\"meter\", \"metre\"), \"not meters\"\n",
    "        return v[\"has_numeric_value\"]\n",
    "    assert isinstance(v, int | float), \"not a number\"\n",
    "    return v\n",
    "\n",
    "\n",
    "def ensure_gold_sequencing_project_identifiers(v, d):\n",
    "    if not isinstance(v, list):\n",
    "        raise Exception(\"list expected\")\n",
    "    if \"gold_sequencing_project_identifiers\" not in d:\n",
    "        return v\n",
    "\n",
    "    rv = []\n",
    "    for elt in v:\n",
    "        if elt.lower().startswith(\"gold:gb\"):\n",
    "            if d[\"id\"].lower().startswith(\"gold:gp\"):\n",
    "                rv.append(d[\"id\"])\n",
    "            else:\n",
    "                raise Exception(\"no avaialable substitute\")\n",
    "        else:\n",
    "            rv.append(elt)\n",
    "    return {\"gold_sequencing_project_identifiers\": rv}\n",
    "\n",
    "\n",
    "def rename_num_tRNA(v):\n",
    "    return [\n",
    "        change_fieldname(elt, \"num_tRNA\", \"num_t_rna\")\n",
    "        for elt in v if \"num_tRNA\" in elt\n",
    "    ]\n",
    "\n",
    "\n",
    "fieldname_replacements = {\n",
    "    \"biosample_set\": {\n",
    "        \"INSDC_biosample_identifiers\": \"insdc_biosample_identifiers\",\n",
    "        # # \"part_of\": \"sample_link\",\n",
    "        \"identifier\": \"samp_name\",\n",
    "        \"GOLD_sample_identifiers\": \"gold_biosample_identifiers\",\n",
    "        \"env_broad_scale\": [ensure_term_id, remove_type_field],\n",
    "        \"env_local_scale\": [ensure_term_id, remove_type_field],\n",
    "        \"env_medium\": [ensure_term_id, remove_type_field],\n",
    "        # depends on `dict`s being ordered in python 3.8+\n",
    "        \"gold_biosample_identifiers\": [uppercase_curie_prefixes],\n",
    "        \"depth2\": [ensure_depth_via_depth2],\n",
    "        \"lat_lon\": [remove_type_field],\n",
    "        \"elev\": [ensure_number_in_meters],\n",
    "        \"insdc_biosample_identifiers\": [lowercase_curie_prefixes],\n",
    "    },\n",
    "    \"study_set\": {\n",
    "        \"GOLD_study_identifiers\": \"gold_study_identifiers\",\n",
    "        \"gold_study_identifiers\": [ensure_curies, uppercase_curie_prefixes],\n",
    "    },\n",
    "    \"omics_processing_set\": {\n",
    "        \"GOLD_sequencing_project_identifiers\": \"gold_sequencing_project_identifiers\",\n",
    "        \"gold_sequencing_project_identifiers\": [\n",
    "            ensure_gold_sequencing_project_identifiers,\n",
    "            uppercase_curie_prefixes,\n",
    "        ],\n",
    "    },\n",
    "    \"mags_activity_set\": {\n",
    "        \"mags_list\": [rename_num_tRNA],\n",
    "        \"lowDepth_contig_num\": \"low_depth_contig_num\",\n",
    "    },\n",
    "    \"metagenome_assembly_set\": {\n",
    "        \"ctg_L50\": \"ctg_l50\",\n",
    "        \"ctg_L90\": \"ctg_l90\",\n",
    "        \"ctg_N50\": \"ctg_n50\",\n",
    "        \"ctg_N90\": \"ctg_n90\",\n",
    "        \"scaf_L50\": \"scaf_l50\",\n",
    "        \"scaf_L90\": \"scaf_l90\",\n",
    "        \"scaf_N50\": \"scaf_n50\",\n",
    "        \"scaf_N90\": \"scaf_n90\",\n",
    "        \"scaf_l_gt50K\": \"scaf_l_gt50k\",\n",
    "        \"scaf_n_gt50K\": \"scaf_n_gt50k\",\n",
    "        \"scaf_pct_gt50K\": \"scaf_pct_gt50k\",\n",
    "    },\n",
    "    # \"read_qc_analysis_activity_set\": {\n",
    "    #     \"output_read_bases\": \"output_base_count\",\n",
    "    #     \"input_read_bases\": \"input_base_count\",\n",
    "    # },\n",
    "}\n",
    "\n",
    "biosample_id_routing = {\n",
    "    \"emsl\": \"emsl_biosample_identifiers\",\n",
    "    \"gold\": \"gold_biosample_identifiers\",\n",
    "    \"igsn\": \"igsn_biosample_identifiers\",\n",
    "    \"img.taxon\": \"img_identifiers\"\n",
    "}\n",
    "\n",
    "\n",
    "def change_fieldname(d, fieldname, replacement):\n",
    "    return dissoc(assoc(d, replacement, d[fieldname]), fieldname)\n",
    "\n",
    "\n",
    "def replace_fields(d, context):\n",
    "    assert \"collection_name\" in context\n",
    "    for fieldname, replacement in fieldname_replacements[context[\"collection_name\"]].items():\n",
    "        if fieldname in d:\n",
    "            if isinstance(replacement, list):\n",
    "                assert all(callable(r) for r in replacement), \"replacement-list must be all functions\"\n",
    "                for rfun in replacement:\n",
    "                    n_params = len(signature(rfun).parameters)\n",
    "                    if n_params == 1:\n",
    "                        d = assoc(d, fieldname, rfun(d[fieldname]))\n",
    "                    elif n_params == 2:\n",
    "                        mapping_updates = rfun(d[fieldname], d)\n",
    "                        for k, v in mapping_updates.items():\n",
    "                            d = dissoc(d, k) if v is None else assoc(d, k, v)\n",
    "                    else:\n",
    "                        raise Exception(\"replaced fn takes too many parameters\")\n",
    "            else:\n",
    "                d = change_fieldname(d, fieldname, replacement)\n",
    "    return d\n",
    "\n",
    "\n",
    "def iter_validate(mdb=None, collection_name=None, limit=0, xform=None):\n",
    "    print(f\"validating {collection_name}...\")\n",
    "    for d in mdb_src[collection_name].find(limit=limit):\n",
    "        d = strip_oid(d)\n",
    "        if xform:\n",
    "            d = xform(d, {\"collection_name\": collection_name})\n",
    "        try:\n",
    "            _ = nmdc_jsonschema_validate({collection_name: [d]})\n",
    "        except fastjsonschema.JsonSchemaException as e:\n",
    "            print(d[\"id\"])\n",
    "            print(e)\n",
    "            #pprint(d)\n",
    "            #raise e\n",
    "            return"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64c4a6bf-04c2-42aa-9019-86b127d3aab0",
   "metadata": {},
   "source": [
    "# replacements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f8b8690f-e4b8-4e46-b0d6-fe57125d5f4b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['biosample_set', 'data_object_set', 'mags_activity_set', 'metabolomics_analysis_activity_set', 'metagenome_annotation_activity_set', 'metagenome_assembly_set', 'metaproteomics_analysis_activity_set', 'metatranscriptome_activity_set', 'nom_analysis_activity_set', 'omics_processing_set', 'read_qc_analysis_activity_set', 'study_set']\n",
      "validating biosample_set...\n",
      "validating data_object_set...\n",
      "validating mags_activity_set...\n",
      "validating metabolomics_analysis_activity_set...\n",
      "validating metagenome_annotation_activity_set...\n",
      "validating metagenome_assembly_set...\n",
      "validating metaproteomics_analysis_activity_set...\n",
      "validating metatranscriptome_activity_set...\n",
      "validating nom_analysis_activity_set...\n",
      "validating omics_processing_set...\n",
      "validating read_qc_analysis_activity_set...\n",
      "validating study_set...\n",
      "Took approx. 95.99997806549072 seconds\n"
     ]
    }
   ],
   "source": [
    "collections = present_src_collections(mdb_src)\n",
    "print(collections)\n",
    "\n",
    "tic = time.time()\n",
    "\n",
    "for coll_name in collections:\n",
    "    if coll_name in fieldname_replacements:\n",
    "        iter_validate(mdb_src, coll_name, xform=replace_fields)\n",
    "    else:\n",
    "        iter_validate(mdb_src, coll_name, xform=None)\n",
    "\n",
    "print(f\"Took approx. {time.time() - tic} seconds\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63783aff-a2ee-4560-a0bd-2cdf37f223b2",
   "metadata": {},
   "source": [
    "## mongodump and mongorestore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4cfe089f-a005-4012-b7c0-a76df4aa906e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ensuring ~/nmdcdb-mongodump/2023-03-01 directory for exports\n"
     ]
    }
   ],
   "source": [
    "from datetime import datetime, timezone\n",
    "from pathlib import Path\n",
    "import time\n",
    "\n",
    "today = datetime.now(timezone.utc).strftime(\"%Y-%m-%d\")\n",
    "print(f\"ensuring ~/nmdcdb-mongodump/{today} directory for exports\")\n",
    "today_dir = Path(\"~/nmdcdb-mongodump\").expanduser().joinpath(today)\n",
    "os.makedirs(str(today_dir), exist_ok=True)\n",
    "\n",
    "collections_excluded = (\n",
    "    set(mdb_src.list_collection_names())\n",
    "    - set(collections + [\"functional_annotation_agg\"])\n",
    ")\n",
    "collections_excluded_str = \" \".join(\n",
    "    [\"--excludeCollection=\" + c for c in collections_excluded]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4d8ef1c8-22f1-47d3-ba2f-5a35939f9367",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-03-01T13:54:16.971-0500\twriting nmdc.data_object_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/data_object_set.bson\n",
      "2023-03-01T13:54:17.049-0500\twriting nmdc.read_qc_analysis_activity_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/read_qc_analysis_activity_set.bson\n",
      "2023-03-01T13:54:17.496-0500\twriting nmdc.functional_annotation_agg to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/functional_annotation_agg.bson\n",
      "2023-03-01T13:54:17.499-0500\twriting nmdc.omics_processing_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/omics_processing_set.bson\n",
      "2023-03-01T13:54:18.620-0500\tdone dumping nmdc.read_qc_analysis_activity_set (1058 documents)\n",
      "2023-03-01T13:54:18.624-0500\t[........................]            nmdc.data_object_set    101/40858  (0.2%)\n",
      "2023-03-01T13:54:18.624-0500\t[........................]  nmdc.functional_annotation_agg  101/3046840  (0.0%)\n",
      "2023-03-01T13:54:18.624-0500\t[#.......................]       nmdc.omics_processing_set     101/1927  (5.2%)\n",
      "2023-03-01T13:54:18.624-0500\t\n",
      "2023-03-01T13:54:21.624-0500\t[........................]            nmdc.data_object_set    101/40858  (0.2%)\n",
      "2023-03-01T13:54:21.624-0500\t[........................]  nmdc.functional_annotation_agg  101/3046840  (0.0%)\n",
      "2023-03-01T13:54:21.624-0500\t[#.......................]       nmdc.omics_processing_set     101/1927  (5.2%)\n",
      "2023-03-01T13:54:21.624-0500\t\n",
      "2023-03-01T13:54:22.640-0500\twriting nmdc.metagenome_assembly_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metagenome_assembly_set.bson\n",
      "2023-03-01T13:54:22.747-0500\t[########################]  nmdc.omics_processing_set  1927/1927  (100.0%)\n",
      "2023-03-01T13:54:22.747-0500\tdone dumping nmdc.omics_processing_set (1927 documents)\n",
      "2023-03-01T13:54:24.322-0500\twriting nmdc.nom_analysis_activity_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/nom_analysis_activity_set.bson\n",
      "2023-03-01T13:54:24.624-0500\t[........................]            nmdc.data_object_set    101/40858  (0.2%)\n",
      "2023-03-01T13:54:24.624-0500\t[........................]  nmdc.functional_annotation_agg  101/3046840  (0.0%)\n",
      "2023-03-01T13:54:24.624-0500\t[........................]    nmdc.metagenome_assembly_set       0/1056  (0.0%)\n",
      "2023-03-01T13:54:24.624-0500\t\n",
      "2023-03-01T13:54:27.340-0500\t[########################]  nmdc.metagenome_assembly_set  1056/1056  (100.0%)\n",
      "2023-03-01T13:54:27.340-0500\tdone dumping nmdc.metagenome_assembly_set (1056 documents)\n",
      "2023-03-01T13:54:27.624-0500\t[........................]            nmdc.data_object_set    101/40858   (0.2%)\n",
      "2023-03-01T13:54:27.624-0500\t[........................]  nmdc.functional_annotation_agg  101/3046840   (0.0%)\n",
      "2023-03-01T13:54:27.624-0500\t[##......................]  nmdc.nom_analysis_activity_set      101/838  (12.1%)\n",
      "2023-03-01T13:54:27.624-0500\t\n",
      "2023-03-01T13:54:28.029-0500\twriting nmdc.metagenome_annotation_activity_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metagenome_annotation_activity_set.bson\n",
      "2023-03-01T13:54:28.030-0500\t[########################]  nmdc.nom_analysis_activity_set  838/838  (100.0%)\n",
      "2023-03-01T13:54:28.030-0500\tdone dumping nmdc.nom_analysis_activity_set (838 documents)\n",
      "2023-03-01T13:54:28.712-0500\twriting nmdc.mags_activity_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/mags_activity_set.bson\n",
      "2023-03-01T13:54:29.532-0500\t[########################]  nmdc.data_object_set  40858/40858  (100.0%)\n",
      "2023-03-01T13:54:29.532-0500\tdone dumping nmdc.data_object_set (40858 documents)\n",
      "2023-03-01T13:54:29.892-0500\tdone dumping nmdc.metagenome_annotation_activity_set (818 documents)\n",
      "2023-03-01T13:54:30.006-0500\twriting nmdc.biosample_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/biosample_set.bson\n",
      "2023-03-01T13:54:30.006-0500\twriting nmdc.metabolomics_analysis_activity_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metabolomics_analysis_activity_set.bson\n",
      "2023-03-01T13:54:30.624-0500\t[........................]           nmdc.functional_annotation_agg  110279/3046840   (3.6%)\n",
      "2023-03-01T13:54:30.624-0500\t[###.....................]                   nmdc.mags_activity_set         101/761  (13.3%)\n",
      "2023-03-01T13:54:30.624-0500\t[###.....................]                       nmdc.biosample_set         101/753  (13.4%)\n",
      "2023-03-01T13:54:30.624-0500\t[........................]  nmdc.metabolomics_analysis_activity_set           0/209   (0.0%)\n",
      "2023-03-01T13:54:30.624-0500\t\n",
      "2023-03-01T13:54:30.807-0500\t[########################]  nmdc.mags_activity_set  761/761  (100.0%)\n",
      "2023-03-01T13:54:30.807-0500\tdone dumping nmdc.mags_activity_set (761 documents)\n",
      "2023-03-01T13:54:31.033-0500\t[########################]  nmdc.biosample_set  753/753  (100.0%)\n",
      "2023-03-01T13:54:31.033-0500\tdone dumping nmdc.biosample_set (753 documents)\n",
      "2023-03-01T13:54:31.163-0500\twriting nmdc.metatranscriptome_activity_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metatranscriptome_activity_set.bson\n",
      "2023-03-01T13:54:31.479-0500\twriting nmdc.metaproteomics_analysis_activity_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metaproteomics_analysis_activity_set.bson\n",
      "2023-03-01T13:54:32.080-0500\tdone dumping nmdc.metatranscriptome_activity_set (55 documents)\n",
      "2023-03-01T13:54:32.783-0500\twriting nmdc.study_set to /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/study_set.bson\n",
      "2023-03-01T13:54:33.624-0500\t[........................]             nmdc.functional_annotation_agg  110279/3046840   (3.6%)\n",
      "2023-03-01T13:54:33.624-0500\t[###########.............]    nmdc.metabolomics_analysis_activity_set         101/209  (48.3%)\n",
      "2023-03-01T13:54:33.624-0500\t[........................]  nmdc.metaproteomics_analysis_activity_set            0/52   (0.0%)\n",
      "2023-03-01T13:54:33.624-0500\t\n",
      "2023-03-01T13:54:34.788-0500\tdone dumping nmdc.study_set (9 documents)\n",
      "2023-03-01T13:54:36.294-0500\t[########################]  nmdc.metabolomics_analysis_activity_set  209/209  (100.0%)\n",
      "2023-03-01T13:54:36.295-0500\tdone dumping nmdc.metabolomics_analysis_activity_set (209 documents)\n",
      "2023-03-01T13:54:36.625-0500\t[........................]             nmdc.functional_annotation_agg  110279/3046840  (3.6%)\n",
      "2023-03-01T13:54:36.625-0500\t[........................]  nmdc.metaproteomics_analysis_activity_set            0/52  (0.0%)\n",
      "2023-03-01T13:54:36.625-0500\t\n",
      "2023-03-01T13:54:39.625-0500\t[#.......................]             nmdc.functional_annotation_agg  219943/3046840  (7.2%)\n",
      "2023-03-01T13:54:39.625-0500\t[........................]  nmdc.metaproteomics_analysis_activity_set            0/52  (0.0%)\n",
      "2023-03-01T13:54:39.625-0500\t\n",
      "2023-03-01T13:54:42.625-0500\t[#.......................]             nmdc.functional_annotation_agg  219943/3046840   (7.2%)\n",
      "2023-03-01T13:54:42.625-0500\t[######..................]  nmdc.metaproteomics_analysis_activity_set           15/52  (28.8%)\n",
      "2023-03-01T13:54:42.625-0500\t\n",
      "2023-03-01T13:54:45.625-0500\t[#.......................]             nmdc.functional_annotation_agg  219943/3046840   (7.2%)\n",
      "2023-03-01T13:54:45.625-0500\t[######..................]  nmdc.metaproteomics_analysis_activity_set           15/52  (28.8%)\n",
      "2023-03-01T13:54:45.625-0500\t\n",
      "2023-03-01T13:54:48.625-0500\t[##......................]             nmdc.functional_annotation_agg  329607/3046840  (10.8%)\n",
      "2023-03-01T13:54:48.625-0500\t[#########...............]  nmdc.metaproteomics_analysis_activity_set           21/52  (40.4%)\n",
      "2023-03-01T13:54:48.625-0500\t\n",
      "2023-03-01T13:54:51.625-0500\t[###.....................]             nmdc.functional_annotation_agg  439271/3046840  (14.4%)\n",
      "2023-03-01T13:54:51.625-0500\t[############............]  nmdc.metaproteomics_analysis_activity_set           26/52  (50.0%)\n",
      "2023-03-01T13:54:51.625-0500\t\n",
      "2023-03-01T13:54:54.625-0500\t[###.....................]             nmdc.functional_annotation_agg  439271/3046840  (14.4%)\n",
      "2023-03-01T13:54:54.625-0500\t[############............]  nmdc.metaproteomics_analysis_activity_set           26/52  (50.0%)\n",
      "2023-03-01T13:54:54.625-0500\t\n",
      "2023-03-01T13:54:57.625-0500\t[####....................]             nmdc.functional_annotation_agg  548935/3046840  (18.0%)\n",
      "2023-03-01T13:54:57.625-0500\t[#############...........]  nmdc.metaproteomics_analysis_activity_set           30/52  (57.7%)\n",
      "2023-03-01T13:54:57.625-0500\t\n",
      "2023-03-01T13:55:00.625-0500\t[####....................]             nmdc.functional_annotation_agg  548935/3046840  (18.0%)\n",
      "2023-03-01T13:55:00.625-0500\t[#############...........]  nmdc.metaproteomics_analysis_activity_set           30/52  (57.7%)\n",
      "2023-03-01T13:55:00.625-0500\t\n",
      "2023-03-01T13:55:03.625-0500\t[#####...................]             nmdc.functional_annotation_agg  658599/3046840  (21.6%)\n",
      "2023-03-01T13:55:03.625-0500\t[################........]  nmdc.metaproteomics_analysis_activity_set           35/52  (67.3%)\n",
      "2023-03-01T13:55:03.625-0500\t\n",
      "2023-03-01T13:55:06.626-0500\t[#####...................]             nmdc.functional_annotation_agg  658599/3046840  (21.6%)\n",
      "2023-03-01T13:55:06.626-0500\t[##################......]  nmdc.metaproteomics_analysis_activity_set           40/52  (76.9%)\n",
      "2023-03-01T13:55:06.626-0500\t\n",
      "2023-03-01T13:55:09.625-0500\t[######..................]             nmdc.functional_annotation_agg  768263/3046840  (25.2%)\n",
      "2023-03-01T13:55:09.625-0500\t[##################......]  nmdc.metaproteomics_analysis_activity_set           40/52  (76.9%)\n",
      "2023-03-01T13:55:09.625-0500\t\n",
      "2023-03-01T13:55:12.626-0500\t[######..................]             nmdc.functional_annotation_agg  768263/3046840  (25.2%)\n",
      "2023-03-01T13:55:12.626-0500\t[###################.....]  nmdc.metaproteomics_analysis_activity_set           43/52  (82.7%)\n",
      "2023-03-01T13:55:12.626-0500\t\n",
      "2023-03-01T13:55:15.626-0500\t[######..................]             nmdc.functional_annotation_agg  877927/3046840  (28.8%)\n",
      "2023-03-01T13:55:15.626-0500\t[####################....]  nmdc.metaproteomics_analysis_activity_set           45/52  (86.5%)\n",
      "2023-03-01T13:55:15.626-0500\t\n",
      "2023-03-01T13:55:18.626-0500\t[#######.................]             nmdc.functional_annotation_agg  987591/3046840  (32.4%)\n",
      "2023-03-01T13:55:18.626-0500\t[#####################...]  nmdc.metaproteomics_analysis_activity_set           46/52  (88.5%)\n",
      "2023-03-01T13:55:18.626-0500\t\n",
      "2023-03-01T13:55:21.626-0500\t[#######.................]             nmdc.functional_annotation_agg  987591/3046840  (32.4%)\n",
      "2023-03-01T13:55:21.626-0500\t[#####################...]  nmdc.metaproteomics_analysis_activity_set           47/52  (90.4%)\n",
      "2023-03-01T13:55:21.626-0500\t\n",
      "2023-03-01T13:55:24.626-0500\t[#######.................]             nmdc.functional_annotation_agg  987591/3046840  (32.4%)\n",
      "2023-03-01T13:55:24.626-0500\t[#####################...]  nmdc.metaproteomics_analysis_activity_set           47/52  (90.4%)\n",
      "2023-03-01T13:55:24.626-0500\t\n",
      "2023-03-01T13:55:27.626-0500\t[########................]             nmdc.functional_annotation_agg  1097255/3046840  (36.0%)\n",
      "2023-03-01T13:55:27.626-0500\t[######################..]  nmdc.metaproteomics_analysis_activity_set            48/52  (92.3%)\n",
      "2023-03-01T13:55:27.626-0500\t\n",
      "2023-03-01T13:55:30.626-0500\t[########................]             nmdc.functional_annotation_agg  1141054/3046840  (37.5%)\n",
      "2023-03-01T13:55:30.626-0500\t[######################..]  nmdc.metaproteomics_analysis_activity_set            49/52  (94.2%)\n",
      "2023-03-01T13:55:30.626-0500\t\n",
      "2023-03-01T13:55:33.626-0500\t[#########...............]             nmdc.functional_annotation_agg  1206919/3046840  (39.6%)\n",
      "2023-03-01T13:55:33.626-0500\t[######################..]  nmdc.metaproteomics_analysis_activity_set            49/52  (94.2%)\n",
      "2023-03-01T13:55:33.626-0500\t\n",
      "2023-03-01T13:55:36.626-0500\t[##########..............]             nmdc.functional_annotation_agg  1316583/3046840  (43.2%)\n",
      "2023-03-01T13:55:36.626-0500\t[#######################.]  nmdc.metaproteomics_analysis_activity_set            50/52  (96.2%)\n",
      "2023-03-01T13:55:36.626-0500\t\n",
      "2023-03-01T13:55:39.627-0500\t[##########..............]             nmdc.functional_annotation_agg  1316583/3046840  (43.2%)\n",
      "2023-03-01T13:55:39.627-0500\t[#######################.]  nmdc.metaproteomics_analysis_activity_set            50/52  (96.2%)\n",
      "2023-03-01T13:55:39.627-0500\t\n",
      "2023-03-01T13:55:42.626-0500\t[##########..............]             nmdc.functional_annotation_agg  1316583/3046840  (43.2%)\n",
      "2023-03-01T13:55:42.626-0500\t[#######################.]  nmdc.metaproteomics_analysis_activity_set            51/52  (98.1%)\n",
      "2023-03-01T13:55:42.626-0500\t\n",
      "2023-03-01T13:55:45.626-0500\t[###########.............]             nmdc.functional_annotation_agg  1426247/3046840  (46.8%)\n",
      "2023-03-01T13:55:45.626-0500\t[#######################.]  nmdc.metaproteomics_analysis_activity_set            51/52  (98.1%)\n",
      "2023-03-01T13:55:45.626-0500\t\n",
      "2023-03-01T13:55:45.659-0500\t[########################]  nmdc.metaproteomics_analysis_activity_set  52/52  (100.0%)\n",
      "2023-03-01T13:55:45.659-0500\tdone dumping nmdc.metaproteomics_analysis_activity_set (52 documents)\n",
      "2023-03-01T13:55:48.626-0500\t[############............]  nmdc.functional_annotation_agg  1535911/3046840  (50.4%)\n",
      "2023-03-01T13:55:51.626-0500\t[############............]  nmdc.functional_annotation_agg  1645575/3046840  (54.0%)\n",
      "2023-03-01T13:55:54.627-0500\t[############............]  nmdc.functional_annotation_agg  1645575/3046840  (54.0%)\n",
      "2023-03-01T13:55:57.627-0500\t[#############...........]  nmdc.functional_annotation_agg  1755239/3046840  (57.6%)\n",
      "2023-03-01T13:56:00.627-0500\t[##############..........]  nmdc.functional_annotation_agg  1864903/3046840  (61.2%)\n",
      "2023-03-01T13:56:03.627-0500\t[###############.........]  nmdc.functional_annotation_agg  1974567/3046840  (64.8%)\n",
      "2023-03-01T13:56:06.627-0500\t[################........]  nmdc.functional_annotation_agg  2084231/3046840  (68.4%)\n",
      "2023-03-01T13:56:09.627-0500\t[#################.......]  nmdc.functional_annotation_agg  2193895/3046840  (72.0%)\n",
      "2023-03-01T13:56:12.627-0500\t[##################......]  nmdc.functional_annotation_agg  2303559/3046840  (75.6%)\n",
      "2023-03-01T13:56:15.627-0500\t[###################.....]  nmdc.functional_annotation_agg  2413223/3046840  (79.2%)\n",
      "2023-03-01T13:56:18.627-0500\t[###################.....]  nmdc.functional_annotation_agg  2522887/3046840  (82.8%)\n",
      "2023-03-01T13:56:21.627-0500\t[#####################...]  nmdc.functional_annotation_agg  2742215/3046840  (90.0%)\n",
      "2023-03-01T13:56:24.628-0500\t[#####################...]  nmdc.functional_annotation_agg  2742215/3046840  (90.0%)\n",
      "2023-03-01T13:56:27.628-0500\t[######################..]  nmdc.functional_annotation_agg  2851879/3046840  (93.6%)\n",
      "2023-03-01T13:56:30.628-0500\t[#######################.]  nmdc.functional_annotation_agg  2961543/3046840  (97.2%)\n",
      "2023-03-01T13:56:33.628-0500\t[#######################.]  nmdc.functional_annotation_agg  2961543/3046840  (97.2%)\n",
      "2023-03-01T13:56:34.413-0500\t[########################]  nmdc.functional_annotation_agg  3046840/3046840  (100.0%)\n",
      "2023-03-01T13:56:34.413-0500\tdone dumping nmdc.functional_annotation_agg (3046840 documents)\n",
      "Took approx. 142.14642596244812 seconds\n"
     ]
    }
   ],
   "source": [
    "tic = time.time()\n",
    "\n",
    "!mongodump --host {os.getenv(\"MONGO_HOST\")} \\\n",
    "     -u {os.getenv(\"MONGO_READONLY_USERNAME\")} \\\n",
    "     -p {os.getenv(\"MONGO_READONLY_PASSWORD\")} \\\n",
    "     --authenticationDatabase admin \\\n",
    "     -d {os.getenv(\"MONGO_DBNAME\")} \\\n",
    "     {collections_excluded_str} \\\n",
    "     -o {today_dir}\n",
    "\n",
    "print(f\"Took approx. {time.time() - tic} seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a230ea80-3993-4678-b8f7-d297ccdd6fc2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-03-01T13:56:39.906-0500\tThe --db and --collection flags are deprecated for this use-case; please use --nsInclude instead, i.e. with --nsInclude=${DATABASE}.${COLLECTION}\n",
      "2023-03-01T13:56:39.907-0500\tbuilding a list of collections to restore from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc dir\n",
      "2023-03-01T13:56:39.908-0500\treading metadata for nmdc-next.functional_annotation_agg from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/functional_annotation_agg.metadata.json\n",
      "2023-03-01T13:56:39.909-0500\treading metadata for nmdc-next.mags_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/mags_activity_set.metadata.json\n",
      "2023-03-01T13:56:39.909-0500\treading metadata for nmdc-next.metagenome_annotation_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metagenome_annotation_activity_set.metadata.json\n",
      "2023-03-01T13:56:39.909-0500\treading metadata for nmdc-next.metatranscriptome_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metatranscriptome_activity_set.metadata.json\n",
      "2023-03-01T13:56:39.909-0500\treading metadata for nmdc-next.omics_processing_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/omics_processing_set.metadata.json\n",
      "2023-03-01T13:56:39.909-0500\treading metadata for nmdc-next.read_qc_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/read_qc_analysis_activity_set.metadata.json\n",
      "2023-03-01T13:56:39.909-0500\treading metadata for nmdc-next.biosample_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/biosample_set.metadata.json\n",
      "2023-03-01T13:56:39.909-0500\treading metadata for nmdc-next.data_object_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/data_object_set.metadata.json\n",
      "2023-03-01T13:56:39.910-0500\treading metadata for nmdc-next.metaproteomics_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metaproteomics_analysis_activity_set.metadata.json\n",
      "2023-03-01T13:56:39.910-0500\treading metadata for nmdc-next.nom_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/nom_analysis_activity_set.metadata.json\n",
      "2023-03-01T13:56:39.910-0500\treading metadata for nmdc-next.study_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/study_set.metadata.json\n",
      "2023-03-01T13:56:39.910-0500\treading metadata for nmdc-next.metabolomics_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metabolomics_analysis_activity_set.metadata.json\n",
      "2023-03-01T13:56:39.910-0500\treading metadata for nmdc-next.metagenome_assembly_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metagenome_assembly_set.metadata.json\n",
      "2023-03-01T13:56:39.991-0500\tdropping collection nmdc-next.data_object_set before restoring\n",
      "2023-03-01T13:56:39.991-0500\tdropping collection nmdc-next.functional_annotation_agg before restoring\n",
      "2023-03-01T13:56:39.991-0500\tdropping collection nmdc-next.metaproteomics_analysis_activity_set before restoring\n",
      "2023-03-01T13:56:39.991-0500\tdropping collection nmdc-next.metabolomics_analysis_activity_set before restoring\n",
      "2023-03-01T13:56:40.570-0500\trestoring nmdc-next.functional_annotation_agg from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/functional_annotation_agg.bson\n",
      "2023-03-01T13:56:40.579-0500\trestoring nmdc-next.data_object_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/data_object_set.bson\n",
      "2023-03-01T13:56:40.622-0500\trestoring nmdc-next.metaproteomics_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metaproteomics_analysis_activity_set.bson\n",
      "2023-03-01T13:56:40.680-0500\trestoring nmdc-next.metabolomics_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metabolomics_analysis_activity_set.bson\n",
      "2023-03-01T13:56:42.621-0500\t[........................]             nmdc-next.functional_annotation_agg    283KB/424MB    (0.1%)\n",
      "2023-03-01T13:56:42.621-0500\t[........................]                       nmdc-next.data_object_set   210KB/14.4MB    (1.4%)\n",
      "2023-03-01T13:56:42.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:56:42.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:56:42.621-0500\t\n",
      "2023-03-01T13:56:45.620-0500\t[........................]             nmdc-next.functional_annotation_agg    425KB/424MB    (0.1%)\n",
      "2023-03-01T13:56:45.620-0500\t[........................]                       nmdc-next.data_object_set   418KB/14.4MB    (2.8%)\n",
      "2023-03-01T13:56:45.620-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:56:45.620-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:56:45.620-0500\t\n",
      "2023-03-01T13:56:48.620-0500\t[........................]             nmdc-next.functional_annotation_agg    850KB/424MB    (0.2%)\n",
      "2023-03-01T13:56:48.620-0500\t[#.......................]                       nmdc-next.data_object_set   990KB/14.4MB    (6.7%)\n",
      "2023-03-01T13:56:48.620-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:56:48.620-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:56:48.620-0500\t\n",
      "2023-03-01T13:56:51.620-0500\t[........................]             nmdc-next.functional_annotation_agg   1.11MB/424MB    (0.3%)\n",
      "2023-03-01T13:56:51.620-0500\t[##......................]                       nmdc-next.data_object_set  1.61MB/14.4MB   (11.2%)\n",
      "2023-03-01T13:56:51.620-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:56:51.620-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:56:51.620-0500\t\n",
      "2023-03-01T13:56:54.620-0500\t[........................]             nmdc-next.functional_annotation_agg   1.80MB/424MB    (0.4%)\n",
      "2023-03-01T13:56:54.620-0500\t[####....................]                       nmdc-next.data_object_set  2.65MB/14.4MB   (18.4%)\n",
      "2023-03-01T13:56:54.620-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:56:54.620-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:56:54.620-0500\t\n",
      "2023-03-01T13:56:57.621-0500\t[........................]             nmdc-next.functional_annotation_agg   2.35MB/424MB    (0.6%)\n",
      "2023-03-01T13:56:57.621-0500\t[#####...................]                       nmdc-next.data_object_set  3.31MB/14.4MB   (23.0%)\n",
      "2023-03-01T13:56:57.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:56:57.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:56:57.621-0500\t\n",
      "2023-03-01T13:57:00.621-0500\t[........................]             nmdc-next.functional_annotation_agg   2.77MB/424MB    (0.7%)\n",
      "2023-03-01T13:57:00.621-0500\t[#######.................]                       nmdc-next.data_object_set  4.20MB/14.4MB   (29.2%)\n",
      "2023-03-01T13:57:00.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:00.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:00.621-0500\t\n",
      "2023-03-01T13:57:03.621-0500\t[........................]             nmdc-next.functional_annotation_agg   3.32MB/424MB    (0.8%)\n",
      "2023-03-01T13:57:03.621-0500\t[########................]                       nmdc-next.data_object_set  5.29MB/14.4MB   (36.8%)\n",
      "2023-03-01T13:57:03.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:03.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:03.621-0500\t\n",
      "2023-03-01T13:57:06.621-0500\t[........................]             nmdc-next.functional_annotation_agg   3.87MB/424MB    (0.9%)\n",
      "2023-03-01T13:57:06.621-0500\t[##########..............]                       nmdc-next.data_object_set  6.04MB/14.4MB   (42.0%)\n",
      "2023-03-01T13:57:06.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:06.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:06.621-0500\t\n",
      "2023-03-01T13:57:09.621-0500\t[........................]             nmdc-next.functional_annotation_agg   4.29MB/424MB    (1.0%)\n",
      "2023-03-01T13:57:09.621-0500\t[###########.............]                       nmdc-next.data_object_set  6.81MB/14.4MB   (47.4%)\n",
      "2023-03-01T13:57:09.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:09.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:09.621-0500\t\n",
      "2023-03-01T13:57:12.621-0500\t[........................]             nmdc-next.functional_annotation_agg   4.98MB/424MB    (1.2%)\n",
      "2023-03-01T13:57:12.621-0500\t[############............]                       nmdc-next.data_object_set  7.56MB/14.4MB   (52.6%)\n",
      "2023-03-01T13:57:12.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:12.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:12.621-0500\t\n",
      "2023-03-01T13:57:15.621-0500\t[........................]             nmdc-next.functional_annotation_agg   5.53MB/424MB    (1.3%)\n",
      "2023-03-01T13:57:15.621-0500\t[##############..........]                       nmdc-next.data_object_set  8.71MB/14.4MB   (60.6%)\n",
      "2023-03-01T13:57:15.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:15.621-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:15.621-0500\t\n",
      "2023-03-01T13:57:18.622-0500\t[........................]             nmdc-next.functional_annotation_agg   5.95MB/424MB    (1.4%)\n",
      "2023-03-01T13:57:18.622-0500\t[###############.........]                       nmdc-next.data_object_set  9.09MB/14.4MB   (63.3%)\n",
      "2023-03-01T13:57:18.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:18.622-0500\t[########################]    nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:18.622-0500\t\n",
      "2023-03-01T13:57:21.081-0500\t[########################]  nmdc-next.metabolomics_analysis_activity_set  15.0MB/15.0MB  (100.0%)\n",
      "2023-03-01T13:57:21.081-0500\tfinished restoring nmdc-next.metabolomics_analysis_activity_set (209 documents, 0 failures)\n",
      "2023-03-01T13:57:21.081-0500\tdropping collection nmdc-next.mags_activity_set before restoring\n",
      "2023-03-01T13:57:21.622-0500\t[........................]             nmdc-next.functional_annotation_agg   6.64MB/424MB   (1.6%)\n",
      "2023-03-01T13:57:21.622-0500\t[#################.......]                       nmdc-next.data_object_set  10.2MB/14.4MB  (70.9%)\n",
      "2023-03-01T13:57:21.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB  (22.2%)\n",
      "2023-03-01T13:57:21.622-0500\t\n",
      "2023-03-01T13:57:21.722-0500\trestoring nmdc-next.mags_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/mags_activity_set.bson\n",
      "2023-03-01T13:57:24.621-0500\t[........................]             nmdc-next.functional_annotation_agg   7.19MB/424MB    (1.7%)\n",
      "2023-03-01T13:57:24.621-0500\t[##################......]                       nmdc-next.data_object_set  11.2MB/14.4MB   (78.0%)\n",
      "2023-03-01T13:57:24.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:24.621-0500\t[########################]                     nmdc-next.mags_activity_set  5.04MB/5.04MB  (100.0%)\n",
      "2023-03-01T13:57:24.621-0500\t\n",
      "2023-03-01T13:57:27.621-0500\t[........................]             nmdc-next.functional_annotation_agg   7.61MB/424MB    (1.8%)\n",
      "2023-03-01T13:57:27.621-0500\t[###################.....]                       nmdc-next.data_object_set  11.6MB/14.4MB   (80.4%)\n",
      "2023-03-01T13:57:27.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:27.621-0500\t[########################]                     nmdc-next.mags_activity_set  5.04MB/5.04MB  (100.0%)\n",
      "2023-03-01T13:57:27.621-0500\t\n",
      "2023-03-01T13:57:30.622-0500\t[........................]             nmdc-next.functional_annotation_agg   8.30MB/424MB    (2.0%)\n",
      "2023-03-01T13:57:30.622-0500\t[#####################...]                       nmdc-next.data_object_set  12.7MB/14.4MB   (88.0%)\n",
      "2023-03-01T13:57:30.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:30.622-0500\t[########################]                     nmdc-next.mags_activity_set  5.04MB/5.04MB  (100.0%)\n",
      "2023-03-01T13:57:30.622-0500\t\n",
      "2023-03-01T13:57:33.622-0500\t[........................]             nmdc-next.functional_annotation_agg   8.85MB/424MB    (2.1%)\n",
      "2023-03-01T13:57:33.622-0500\t[######################..]                       nmdc-next.data_object_set  13.4MB/14.4MB   (93.0%)\n",
      "2023-03-01T13:57:33.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:33.622-0500\t[########################]                     nmdc-next.mags_activity_set  5.04MB/5.04MB  (100.0%)\n",
      "2023-03-01T13:57:33.622-0500\t\n",
      "2023-03-01T13:57:33.655-0500\t[########################]  nmdc-next.mags_activity_set  5.04MB/5.04MB  (100.0%)\n",
      "2023-03-01T13:57:33.655-0500\tfinished restoring nmdc-next.mags_activity_set (761 documents, 0 failures)\n",
      "2023-03-01T13:57:33.655-0500\tdropping collection nmdc-next.metagenome_assembly_set before restoring\n",
      "2023-03-01T13:57:34.306-0500\trestoring nmdc-next.metagenome_assembly_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metagenome_assembly_set.bson\n",
      "2023-03-01T13:57:36.621-0500\t[........................]             nmdc-next.functional_annotation_agg   9.40MB/424MB    (2.2%)\n",
      "2023-03-01T13:57:36.621-0500\t[########################]                       nmdc-next.data_object_set  14.4MB/14.4MB  (100.0%)\n",
      "2023-03-01T13:57:36.621-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:36.621-0500\t[######################..]               nmdc-next.metagenome_assembly_set  1.06MB/1.11MB   (95.3%)\n",
      "2023-03-01T13:57:36.621-0500\t\n",
      "2023-03-01T13:57:37.577-0500\t[########################]  nmdc-next.data_object_set  14.4MB/14.4MB  (100.0%)\n",
      "2023-03-01T13:57:37.577-0500\tfinished restoring nmdc-next.data_object_set (40858 documents, 0 failures)\n",
      "2023-03-01T13:57:37.577-0500\tdropping collection nmdc-next.omics_processing_set before restoring\n",
      "2023-03-01T13:57:37.754-0500\t[########################]  nmdc-next.metagenome_assembly_set  1.11MB/1.11MB  (100.0%)\n",
      "2023-03-01T13:57:37.754-0500\tfinished restoring nmdc-next.metagenome_assembly_set (1056 documents, 0 failures)\n",
      "2023-03-01T13:57:37.754-0500\tdropping collection nmdc-next.metagenome_annotation_activity_set before restoring\n",
      "2023-03-01T13:57:38.225-0500\trestoring nmdc-next.omics_processing_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/omics_processing_set.bson\n",
      "2023-03-01T13:57:38.495-0500\trestoring nmdc-next.metagenome_annotation_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metagenome_annotation_activity_set.bson\n",
      "2023-03-01T13:57:39.622-0500\t[........................]             nmdc-next.functional_annotation_agg   10.1MB/424MB    (2.4%)\n",
      "2023-03-01T13:57:39.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:39.622-0500\t[########################]                  nmdc-next.omics_processing_set  1.05MB/1.05MB  (100.0%)\n",
      "2023-03-01T13:57:39.622-0500\t[########################]    nmdc-next.metagenome_annotation_activity_set  1.03MB/1.03MB  (100.0%)\n",
      "2023-03-01T13:57:39.622-0500\t\n",
      "2023-03-01T13:57:40.990-0500\t[########################]  nmdc-next.omics_processing_set  1.05MB/1.05MB  (100.0%)\n",
      "2023-03-01T13:57:40.990-0500\tfinished restoring nmdc-next.omics_processing_set (1927 documents, 0 failures)\n",
      "2023-03-01T13:57:40.990-0500\tdropping collection nmdc-next.biosample_set before restoring\n",
      "2023-03-01T13:57:40.998-0500\t[########################]  nmdc-next.metagenome_annotation_activity_set  1.03MB/1.03MB  (100.0%)\n",
      "2023-03-01T13:57:40.998-0500\tfinished restoring nmdc-next.metagenome_annotation_activity_set (818 documents, 0 failures)\n",
      "2023-03-01T13:57:40.998-0500\tdropping collection nmdc-next.read_qc_analysis_activity_set before restoring\n",
      "2023-03-01T13:57:41.641-0500\trestoring nmdc-next.biosample_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/biosample_set.bson\n",
      "2023-03-01T13:57:41.778-0500\trestoring nmdc-next.read_qc_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/read_qc_analysis_activity_set.bson\n",
      "2023-03-01T13:57:42.622-0500\t[........................]             nmdc-next.functional_annotation_agg   10.7MB/424MB    (2.5%)\n",
      "2023-03-01T13:57:42.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:42.622-0500\t[########################]                         nmdc-next.biosample_set  1008KB/1008KB  (100.0%)\n",
      "2023-03-01T13:57:42.622-0500\t[######################..]         nmdc-next.read_qc_analysis_activity_set    658KB/696KB   (94.5%)\n",
      "2023-03-01T13:57:42.622-0500\t\n",
      "2023-03-01T13:57:44.172-0500\t[########################]  nmdc-next.biosample_set  1008KB/1008KB  (100.0%)\n",
      "2023-03-01T13:57:44.172-0500\tfinished restoring nmdc-next.biosample_set (753 documents, 0 failures)\n",
      "2023-03-01T13:57:44.172-0500\tdropping collection nmdc-next.nom_analysis_activity_set before restoring\n",
      "2023-03-01T13:57:44.598-0500\t[########################]  nmdc-next.read_qc_analysis_activity_set  696KB/696KB  (100.0%)\n",
      "2023-03-01T13:57:44.598-0500\tfinished restoring nmdc-next.read_qc_analysis_activity_set (1058 documents, 0 failures)\n",
      "2023-03-01T13:57:44.598-0500\tdropping collection nmdc-next.study_set before restoring\n",
      "2023-03-01T13:57:45.159-0500\trestoring nmdc-next.nom_analysis_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/nom_analysis_activity_set.bson\n",
      "2023-03-01T13:57:45.446-0500\trestoring nmdc-next.study_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/study_set.bson\n",
      "2023-03-01T13:57:45.622-0500\t[........................]             nmdc-next.functional_annotation_agg   11.2MB/424MB    (2.6%)\n",
      "2023-03-01T13:57:45.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set   43.9MB/198MB   (22.2%)\n",
      "2023-03-01T13:57:45.622-0500\t[########################]             nmdc-next.nom_analysis_activity_set    356KB/356KB  (100.0%)\n",
      "2023-03-01T13:57:45.622-0500\t[########################]                             nmdc-next.study_set  56.3KB/56.3KB  (100.0%)\n",
      "2023-03-01T13:57:45.622-0500\t\n",
      "2023-03-01T13:57:45.838-0500\t[########################]  nmdc-next.study_set  56.3KB/56.3KB  (100.0%)\n",
      "2023-03-01T13:57:45.838-0500\tfinished restoring nmdc-next.study_set (9 documents, 0 failures)\n",
      "2023-03-01T13:57:45.838-0500\tdropping collection nmdc-next.metatranscriptome_activity_set before restoring\n",
      "2023-03-01T13:57:46.201-0500\t[########################]  nmdc-next.nom_analysis_activity_set  356KB/356KB  (100.0%)\n",
      "2023-03-01T13:57:46.201-0500\tfinished restoring nmdc-next.nom_analysis_activity_set (838 documents, 0 failures)\n",
      "2023-03-01T13:57:46.776-0500\trestoring nmdc-next.metatranscriptome_activity_set from /Users/dwinston/nmdcdb-mongodump/2023-03-01/nmdc/metatranscriptome_activity_set.bson\n",
      "2023-03-01T13:57:47.114-0500\tfinished restoring nmdc-next.metatranscriptome_activity_set (55 documents, 0 failures)\n",
      "2023-03-01T13:57:48.622-0500\t[........................]             nmdc-next.functional_annotation_agg  12.0MB/424MB   (2.8%)\n",
      "2023-03-01T13:57:48.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set  43.9MB/198MB  (22.2%)\n",
      "2023-03-01T13:57:48.622-0500\t\n",
      "2023-03-01T13:57:51.622-0500\t[........................]             nmdc-next.functional_annotation_agg  12.9MB/424MB   (3.0%)\n",
      "2023-03-01T13:57:51.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set  43.9MB/198MB  (22.2%)\n",
      "2023-03-01T13:57:51.622-0500\t\n",
      "2023-03-01T13:57:54.622-0500\t[........................]             nmdc-next.functional_annotation_agg  13.7MB/424MB   (3.2%)\n",
      "2023-03-01T13:57:54.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set  43.9MB/198MB  (22.2%)\n",
      "2023-03-01T13:57:54.622-0500\t\n",
      "2023-03-01T13:57:57.623-0500\t[........................]             nmdc-next.functional_annotation_agg  14.5MB/424MB   (3.4%)\n",
      "2023-03-01T13:57:57.623-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set  43.9MB/198MB  (22.2%)\n",
      "2023-03-01T13:57:57.623-0500\t\n",
      "2023-03-01T13:58:00.622-0500\t[........................]             nmdc-next.functional_annotation_agg  15.5MB/424MB   (3.7%)\n",
      "2023-03-01T13:58:00.622-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set  43.9MB/198MB  (22.2%)\n",
      "2023-03-01T13:58:00.622-0500\t\n",
      "2023-03-01T13:58:03.623-0500\t[........................]             nmdc-next.functional_annotation_agg  16.1MB/424MB   (3.8%)\n",
      "2023-03-01T13:58:03.623-0500\t[#####...................]  nmdc-next.metaproteomics_analysis_activity_set  43.9MB/198MB  (22.2%)\n",
      "2023-03-01T13:58:03.623-0500\t\n",
      "2023-03-01T13:58:06.622-0500\t[........................]             nmdc-next.functional_annotation_agg  17.2MB/424MB   (4.1%)\n",
      "2023-03-01T13:58:06.622-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:06.622-0500\t\n",
      "2023-03-01T13:58:09.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  18.0MB/424MB   (4.2%)\n",
      "2023-03-01T13:58:09.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:09.623-0500\t\n",
      "2023-03-01T13:58:12.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  18.7MB/424MB   (4.4%)\n",
      "2023-03-01T13:58:12.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:12.623-0500\t\n",
      "2023-03-01T13:58:15.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  19.6MB/424MB   (4.6%)\n",
      "2023-03-01T13:58:15.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:15.623-0500\t\n",
      "2023-03-01T13:58:18.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  20.4MB/424MB   (4.8%)\n",
      "2023-03-01T13:58:18.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:18.623-0500\t\n",
      "2023-03-01T13:58:21.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  21.2MB/424MB   (5.0%)\n",
      "2023-03-01T13:58:21.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:21.623-0500\t\n",
      "2023-03-01T13:58:24.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  21.9MB/424MB   (5.2%)\n",
      "2023-03-01T13:58:24.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:24.623-0500\t\n",
      "2023-03-01T13:58:27.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  22.8MB/424MB   (5.4%)\n",
      "2023-03-01T13:58:27.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:27.623-0500\t\n",
      "2023-03-01T13:58:30.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  23.5MB/424MB   (5.5%)\n",
      "2023-03-01T13:58:30.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:30.623-0500\t\n",
      "2023-03-01T13:58:33.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  24.4MB/424MB   (5.8%)\n",
      "2023-03-01T13:58:33.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:33.623-0500\t\n",
      "2023-03-01T13:58:36.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  25.4MB/424MB   (6.0%)\n",
      "2023-03-01T13:58:36.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:36.623-0500\t\n",
      "2023-03-01T13:58:39.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  26.1MB/424MB   (6.2%)\n",
      "2023-03-01T13:58:39.624-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:39.624-0500\t\n",
      "2023-03-01T13:58:42.623-0500\t[#.......................]             nmdc-next.functional_annotation_agg  26.9MB/424MB   (6.4%)\n",
      "2023-03-01T13:58:42.623-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:42.623-0500\t\n",
      "2023-03-01T13:58:45.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  27.8MB/424MB   (6.5%)\n",
      "2023-03-01T13:58:45.624-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:45.624-0500\t\n",
      "2023-03-01T13:58:48.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  28.5MB/424MB   (6.7%)\n",
      "2023-03-01T13:58:48.624-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:48.624-0500\t\n",
      "2023-03-01T13:58:51.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  29.2MB/424MB   (6.9%)\n",
      "2023-03-01T13:58:51.624-0500\t[#############...........]  nmdc-next.metaproteomics_analysis_activity_set   109MB/198MB  (55.1%)\n",
      "2023-03-01T13:58:51.624-0500\t\n",
      "2023-03-01T13:58:54.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  30.1MB/424MB    (7.1%)\n",
      "2023-03-01T13:58:54.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:58:54.624-0500\t\n",
      "2023-03-01T13:58:57.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  31.0MB/424MB    (7.3%)\n",
      "2023-03-01T13:58:57.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:58:57.624-0500\t\n",
      "2023-03-01T13:59:00.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  31.5MB/424MB    (7.4%)\n",
      "2023-03-01T13:59:00.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:00.624-0500\t\n",
      "2023-03-01T13:59:03.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  32.5MB/424MB    (7.7%)\n",
      "2023-03-01T13:59:03.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:03.624-0500\t\n",
      "2023-03-01T13:59:06.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  33.3MB/424MB    (7.9%)\n",
      "2023-03-01T13:59:06.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:06.624-0500\t\n",
      "2023-03-01T13:59:09.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  34.2MB/424MB    (8.1%)\n",
      "2023-03-01T13:59:09.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:09.624-0500\t\n",
      "2023-03-01T13:59:12.624-0500\t[#.......................]             nmdc-next.functional_annotation_agg  34.9MB/424MB    (8.2%)\n",
      "2023-03-01T13:59:12.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:12.625-0500\t\n",
      "2023-03-01T13:59:15.624-0500\t[##......................]             nmdc-next.functional_annotation_agg  35.7MB/424MB    (8.4%)\n",
      "2023-03-01T13:59:15.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:15.624-0500\t\n",
      "2023-03-01T13:59:18.624-0500\t[##......................]             nmdc-next.functional_annotation_agg  36.5MB/424MB    (8.6%)\n",
      "2023-03-01T13:59:18.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:18.624-0500\t\n",
      "2023-03-01T13:59:21.624-0500\t[##......................]             nmdc-next.functional_annotation_agg  37.4MB/424MB    (8.8%)\n",
      "2023-03-01T13:59:21.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:21.624-0500\t\n",
      "2023-03-01T13:59:24.624-0500\t[##......................]             nmdc-next.functional_annotation_agg  38.2MB/424MB    (9.0%)\n",
      "2023-03-01T13:59:24.624-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:24.624-0500\t\n",
      "2023-03-01T13:59:27.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  39.0MB/424MB    (9.2%)\n",
      "2023-03-01T13:59:27.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:27.625-0500\t\n",
      "2023-03-01T13:59:30.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  39.8MB/424MB    (9.4%)\n",
      "2023-03-01T13:59:30.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:30.625-0500\t\n",
      "2023-03-01T13:59:33.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  40.7MB/424MB    (9.6%)\n",
      "2023-03-01T13:59:33.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:33.625-0500\t\n",
      "2023-03-01T13:59:36.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  41.6MB/424MB    (9.8%)\n",
      "2023-03-01T13:59:36.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:36.625-0500\t\n",
      "2023-03-01T13:59:39.626-0500\t[##......................]             nmdc-next.functional_annotation_agg  42.4MB/424MB   (10.0%)\n",
      "2023-03-01T13:59:39.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:39.626-0500\t\n",
      "2023-03-01T13:59:42.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  43.4MB/424MB   (10.2%)\n",
      "2023-03-01T13:59:42.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:42.625-0500\t\n",
      "2023-03-01T13:59:45.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  44.1MB/424MB   (10.4%)\n",
      "2023-03-01T13:59:45.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:45.625-0500\t\n",
      "2023-03-01T13:59:48.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  45.0MB/424MB   (10.6%)\n",
      "2023-03-01T13:59:48.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:48.625-0500\t\n",
      "2023-03-01T13:59:51.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  45.9MB/424MB   (10.8%)\n",
      "2023-03-01T13:59:51.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:51.625-0500\t\n",
      "2023-03-01T13:59:54.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  46.6MB/424MB   (11.0%)\n",
      "2023-03-01T13:59:54.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:54.625-0500\t\n",
      "2023-03-01T13:59:57.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  47.3MB/424MB   (11.1%)\n",
      "2023-03-01T13:59:57.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T13:59:57.625-0500\t\n",
      "2023-03-01T14:00:00.625-0500\t[##......................]             nmdc-next.functional_annotation_agg  48.1MB/424MB   (11.3%)\n",
      "2023-03-01T14:00:00.625-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:00.625-0500\t\n",
      "2023-03-01T14:00:03.626-0500\t[##......................]             nmdc-next.functional_annotation_agg  48.9MB/424MB   (11.5%)\n",
      "2023-03-01T14:00:03.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:03.626-0500\t\n",
      "2023-03-01T14:00:06.626-0500\t[##......................]             nmdc-next.functional_annotation_agg  49.6MB/424MB   (11.7%)\n",
      "2023-03-01T14:00:06.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:06.626-0500\t\n",
      "2023-03-01T14:00:09.626-0500\t[##......................]             nmdc-next.functional_annotation_agg  50.3MB/424MB   (11.9%)\n",
      "2023-03-01T14:00:09.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:09.626-0500\t\n",
      "2023-03-01T14:00:12.626-0500\t[##......................]             nmdc-next.functional_annotation_agg  51.2MB/424MB   (12.1%)\n",
      "2023-03-01T14:00:12.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:12.626-0500\t\n",
      "2023-03-01T14:00:15.626-0500\t[##......................]             nmdc-next.functional_annotation_agg  51.7MB/424MB   (12.2%)\n",
      "2023-03-01T14:00:15.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:15.626-0500\t\n",
      "2023-03-01T14:00:18.626-0500\t[##......................]             nmdc-next.functional_annotation_agg  52.6MB/424MB   (12.4%)\n",
      "2023-03-01T14:00:18.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:18.626-0500\t\n",
      "2023-03-01T14:00:21.626-0500\t[###.....................]             nmdc-next.functional_annotation_agg  53.4MB/424MB   (12.6%)\n",
      "2023-03-01T14:00:21.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:21.626-0500\t\n",
      "2023-03-01T14:00:24.626-0500\t[###.....................]             nmdc-next.functional_annotation_agg  54.1MB/424MB   (12.8%)\n",
      "2023-03-01T14:00:24.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:24.626-0500\t\n",
      "2023-03-01T14:00:27.626-0500\t[###.....................]             nmdc-next.functional_annotation_agg  54.9MB/424MB   (12.9%)\n",
      "2023-03-01T14:00:27.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:27.626-0500\t\n",
      "2023-03-01T14:00:30.626-0500\t[###.....................]             nmdc-next.functional_annotation_agg  55.8MB/424MB   (13.1%)\n",
      "2023-03-01T14:00:30.626-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set   198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:30.626-0500\t\n",
      "2023-03-01T14:00:30.834-0500\t[########################]  nmdc-next.metaproteomics_analysis_activity_set  198MB/198MB  (100.0%)\n",
      "2023-03-01T14:00:30.834-0500\tfinished restoring nmdc-next.metaproteomics_analysis_activity_set (52 documents, 0 failures)\n",
      "2023-03-01T14:00:33.626-0500\t[###.....................]  nmdc-next.functional_annotation_agg  58.3MB/424MB  (13.7%)\n",
      "2023-03-01T14:00:36.627-0500\t[###.....................]  nmdc-next.functional_annotation_agg  60.9MB/424MB  (14.4%)\n",
      "2023-03-01T14:00:39.626-0500\t[###.....................]  nmdc-next.functional_annotation_agg  63.3MB/424MB  (14.9%)\n",
      "2023-03-01T14:00:42.627-0500\t[###.....................]  nmdc-next.functional_annotation_agg  65.6MB/424MB  (15.5%)\n",
      "2023-03-01T14:00:45.626-0500\t[###.....................]  nmdc-next.functional_annotation_agg  68.4MB/424MB  (16.1%)\n",
      "2023-03-01T14:00:48.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  70.8MB/424MB  (16.7%)\n",
      "2023-03-01T14:00:51.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  73.3MB/424MB  (17.3%)\n",
      "2023-03-01T14:00:54.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  76.1MB/424MB  (17.9%)\n",
      "2023-03-01T14:00:57.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  78.5MB/424MB  (18.5%)\n",
      "2023-03-01T14:01:00.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  80.4MB/424MB  (19.0%)\n",
      "2023-03-01T14:01:03.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  82.6MB/424MB  (19.5%)\n",
      "2023-03-01T14:01:06.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  85.5MB/424MB  (20.2%)\n",
      "2023-03-01T14:01:09.627-0500\t[####....................]  nmdc-next.functional_annotation_agg  88.2MB/424MB  (20.8%)\n",
      "2023-03-01T14:01:12.627-0500\t[#####...................]  nmdc-next.functional_annotation_agg  90.4MB/424MB  (21.3%)\n",
      "2023-03-01T14:01:15.627-0500\t[#####...................]  nmdc-next.functional_annotation_agg  93.1MB/424MB  (21.9%)\n",
      "2023-03-01T14:01:18.627-0500\t[#####...................]  nmdc-next.functional_annotation_agg  95.9MB/424MB  (22.6%)\n",
      "2023-03-01T14:01:21.628-0500\t[#####...................]  nmdc-next.functional_annotation_agg  98.2MB/424MB  (23.2%)\n",
      "2023-03-01T14:01:24.628-0500\t[#####...................]  nmdc-next.functional_annotation_agg  101MB/424MB  (23.7%)\n",
      "2023-03-01T14:01:27.628-0500\t[#####...................]  nmdc-next.functional_annotation_agg  104MB/424MB  (24.4%)\n",
      "2023-03-01T14:01:30.628-0500\t[#####...................]  nmdc-next.functional_annotation_agg  106MB/424MB  (25.0%)\n",
      "2023-03-01T14:01:33.628-0500\t[######..................]  nmdc-next.functional_annotation_agg  109MB/424MB  (25.6%)\n",
      "2023-03-01T14:01:36.628-0500\t[######..................]  nmdc-next.functional_annotation_agg  111MB/424MB  (26.2%)\n",
      "2023-03-01T14:01:39.628-0500\t[######..................]  nmdc-next.functional_annotation_agg  114MB/424MB  (26.8%)\n",
      "2023-03-01T14:01:42.628-0500\t[######..................]  nmdc-next.functional_annotation_agg  116MB/424MB  (27.4%)\n",
      "2023-03-01T14:01:45.628-0500\t[######..................]  nmdc-next.functional_annotation_agg  119MB/424MB  (28.0%)\n",
      "2023-03-01T14:01:48.628-0500\t[######..................]  nmdc-next.functional_annotation_agg  121MB/424MB  (28.6%)\n",
      "2023-03-01T14:01:51.628-0500\t[#######.................]  nmdc-next.functional_annotation_agg  124MB/424MB  (29.2%)\n",
      "2023-03-01T14:01:54.628-0500\t[#######.................]  nmdc-next.functional_annotation_agg  127MB/424MB  (29.9%)\n",
      "2023-03-01T14:01:57.628-0500\t[#######.................]  nmdc-next.functional_annotation_agg  129MB/424MB  (30.4%)\n",
      "2023-03-01T14:02:00.629-0500\t[#######.................]  nmdc-next.functional_annotation_agg  131MB/424MB  (30.9%)\n",
      "2023-03-01T14:02:03.629-0500\t[#######.................]  nmdc-next.functional_annotation_agg  134MB/424MB  (31.5%)\n",
      "2023-03-01T14:02:06.629-0500\t[#######.................]  nmdc-next.functional_annotation_agg  136MB/424MB  (32.1%)\n",
      "2023-03-01T14:02:09.629-0500\t[#######.................]  nmdc-next.functional_annotation_agg  139MB/424MB  (32.7%)\n",
      "2023-03-01T14:02:12.629-0500\t[########................]  nmdc-next.functional_annotation_agg  142MB/424MB  (33.4%)\n",
      "2023-03-01T14:02:15.629-0500\t[########................]  nmdc-next.functional_annotation_agg  144MB/424MB  (33.9%)\n",
      "2023-03-01T14:02:18.629-0500\t[########................]  nmdc-next.functional_annotation_agg  147MB/424MB  (34.5%)\n",
      "2023-03-01T14:02:21.629-0500\t[########................]  nmdc-next.functional_annotation_agg  149MB/424MB  (35.2%)\n",
      "2023-03-01T14:02:24.629-0500\t[########................]  nmdc-next.functional_annotation_agg  152MB/424MB  (35.8%)\n",
      "2023-03-01T14:02:27.629-0500\t[########................]  nmdc-next.functional_annotation_agg  154MB/424MB  (36.3%)\n",
      "2023-03-01T14:02:30.629-0500\t[########................]  nmdc-next.functional_annotation_agg  157MB/424MB  (36.9%)\n",
      "2023-03-01T14:02:33.630-0500\t[#########...............]  nmdc-next.functional_annotation_agg  159MB/424MB  (37.5%)\n",
      "2023-03-01T14:02:36.629-0500\t[#########...............]  nmdc-next.functional_annotation_agg  162MB/424MB  (38.1%)\n",
      "2023-03-01T14:02:39.630-0500\t[#########...............]  nmdc-next.functional_annotation_agg  164MB/424MB  (38.8%)\n",
      "2023-03-01T14:02:42.630-0500\t[#########...............]  nmdc-next.functional_annotation_agg  167MB/424MB  (39.3%)\n",
      "2023-03-01T14:02:45.630-0500\t[#########...............]  nmdc-next.functional_annotation_agg  169MB/424MB  (39.9%)\n",
      "2023-03-01T14:02:48.630-0500\t[#########...............]  nmdc-next.functional_annotation_agg  172MB/424MB  (40.6%)\n",
      "2023-03-01T14:02:51.630-0500\t[#########...............]  nmdc-next.functional_annotation_agg  174MB/424MB  (41.1%)\n",
      "2023-03-01T14:02:54.630-0500\t[##########..............]  nmdc-next.functional_annotation_agg  177MB/424MB  (41.7%)\n",
      "2023-03-01T14:02:57.630-0500\t[##########..............]  nmdc-next.functional_annotation_agg  180MB/424MB  (42.4%)\n",
      "2023-03-01T14:03:00.630-0500\t[##########..............]  nmdc-next.functional_annotation_agg  182MB/424MB  (42.9%)\n",
      "2023-03-01T14:03:03.630-0500\t[##########..............]  nmdc-next.functional_annotation_agg  184MB/424MB  (43.5%)\n",
      "2023-03-01T14:03:06.630-0500\t[##########..............]  nmdc-next.functional_annotation_agg  187MB/424MB  (44.1%)\n",
      "2023-03-01T14:03:09.630-0500\t[##########..............]  nmdc-next.functional_annotation_agg  190MB/424MB  (44.7%)\n",
      "2023-03-01T14:03:12.630-0500\t[##########..............]  nmdc-next.functional_annotation_agg  192MB/424MB  (45.3%)\n",
      "2023-03-01T14:03:15.630-0500\t[###########.............]  nmdc-next.functional_annotation_agg  195MB/424MB  (46.0%)\n",
      "2023-03-01T14:03:18.631-0500\t[###########.............]  nmdc-next.functional_annotation_agg  198MB/424MB  (46.6%)\n",
      "2023-03-01T14:03:21.631-0500\t[###########.............]  nmdc-next.functional_annotation_agg  200MB/424MB  (47.1%)\n",
      "2023-03-01T14:03:24.631-0500\t[###########.............]  nmdc-next.functional_annotation_agg  203MB/424MB  (47.7%)\n",
      "2023-03-01T14:03:27.631-0500\t[###########.............]  nmdc-next.functional_annotation_agg  205MB/424MB  (48.3%)\n",
      "2023-03-01T14:03:30.631-0500\t[###########.............]  nmdc-next.functional_annotation_agg  207MB/424MB  (48.9%)\n",
      "2023-03-01T14:03:33.631-0500\t[###########.............]  nmdc-next.functional_annotation_agg  210MB/424MB  (49.6%)\n",
      "2023-03-01T14:03:36.631-0500\t[############............]  nmdc-next.functional_annotation_agg  213MB/424MB  (50.1%)\n",
      "2023-03-01T14:03:39.631-0500\t[############............]  nmdc-next.functional_annotation_agg  215MB/424MB  (50.7%)\n",
      "2023-03-01T14:03:42.631-0500\t[############............]  nmdc-next.functional_annotation_agg  218MB/424MB  (51.4%)\n",
      "2023-03-01T14:03:45.631-0500\t[############............]  nmdc-next.functional_annotation_agg  220MB/424MB  (51.9%)\n",
      "2023-03-01T14:03:48.632-0500\t[############............]  nmdc-next.functional_annotation_agg  223MB/424MB  (52.5%)\n",
      "2023-03-01T14:03:51.634-0500\t[############............]  nmdc-next.functional_annotation_agg  225MB/424MB  (53.2%)\n",
      "2023-03-01T14:03:54.632-0500\t[############............]  nmdc-next.functional_annotation_agg  228MB/424MB  (53.8%)\n",
      "2023-03-01T14:03:57.632-0500\t[#############...........]  nmdc-next.functional_annotation_agg  230MB/424MB  (54.3%)\n",
      "2023-03-01T14:04:00.632-0500\t[#############...........]  nmdc-next.functional_annotation_agg  233MB/424MB  (55.0%)\n",
      "2023-03-01T14:04:03.632-0500\t[#############...........]  nmdc-next.functional_annotation_agg  236MB/424MB  (55.6%)\n",
      "2023-03-01T14:04:06.632-0500\t[#############...........]  nmdc-next.functional_annotation_agg  238MB/424MB  (56.2%)\n",
      "2023-03-01T14:04:09.633-0500\t[#############...........]  nmdc-next.functional_annotation_agg  241MB/424MB  (56.9%)\n",
      "2023-03-01T14:04:12.633-0500\t[#############...........]  nmdc-next.functional_annotation_agg  244MB/424MB  (57.5%)\n",
      "2023-03-01T14:04:15.633-0500\t[#############...........]  nmdc-next.functional_annotation_agg  246MB/424MB  (58.0%)\n",
      "2023-03-01T14:04:18.632-0500\t[##############..........]  nmdc-next.functional_annotation_agg  248MB/424MB  (58.4%)\n",
      "2023-03-01T14:04:21.633-0500\t[##############..........]  nmdc-next.functional_annotation_agg  250MB/424MB  (59.0%)\n",
      "2023-03-01T14:04:24.632-0500\t[##############..........]  nmdc-next.functional_annotation_agg  252MB/424MB  (59.5%)\n",
      "2023-03-01T14:04:27.632-0500\t[##############..........]  nmdc-next.functional_annotation_agg  255MB/424MB  (60.2%)\n",
      "2023-03-01T14:04:30.633-0500\t[##############..........]  nmdc-next.functional_annotation_agg  257MB/424MB  (60.6%)\n",
      "2023-03-01T14:04:33.633-0500\t[##############..........]  nmdc-next.functional_annotation_agg  259MB/424MB  (61.0%)\n",
      "2023-03-01T14:04:36.633-0500\t[##############..........]  nmdc-next.functional_annotation_agg  261MB/424MB  (61.5%)\n",
      "2023-03-01T14:04:39.633-0500\t[##############..........]  nmdc-next.functional_annotation_agg  263MB/424MB  (62.1%)\n",
      "2023-03-01T14:04:42.633-0500\t[###############.........]  nmdc-next.functional_annotation_agg  265MB/424MB  (62.6%)\n",
      "2023-03-01T14:04:45.633-0500\t[###############.........]  nmdc-next.functional_annotation_agg  268MB/424MB  (63.3%)\n",
      "2023-03-01T14:04:48.633-0500\t[###############.........]  nmdc-next.functional_annotation_agg  271MB/424MB  (63.8%)\n",
      "2023-03-01T14:04:51.633-0500\t[###############.........]  nmdc-next.functional_annotation_agg  273MB/424MB  (64.4%)\n",
      "2023-03-01T14:04:54.633-0500\t[###############.........]  nmdc-next.functional_annotation_agg  276MB/424MB  (65.0%)\n",
      "2023-03-01T14:04:57.633-0500\t[###############.........]  nmdc-next.functional_annotation_agg  278MB/424MB  (65.6%)\n",
      "2023-03-01T14:05:00.634-0500\t[###############.........]  nmdc-next.functional_annotation_agg  280MB/424MB  (66.1%)\n",
      "2023-03-01T14:05:03.634-0500\t[################........]  nmdc-next.functional_annotation_agg  283MB/424MB  (66.7%)\n",
      "2023-03-01T14:05:06.633-0500\t[################........]  nmdc-next.functional_annotation_agg  286MB/424MB  (67.3%)\n",
      "2023-03-01T14:05:09.634-0500\t[################........]  nmdc-next.functional_annotation_agg  288MB/424MB  (67.9%)\n",
      "2023-03-01T14:05:12.634-0500\t[################........]  nmdc-next.functional_annotation_agg  291MB/424MB  (68.6%)\n",
      "2023-03-01T14:05:15.635-0500\t[################........]  nmdc-next.functional_annotation_agg  293MB/424MB  (69.1%)\n",
      "2023-03-01T14:05:18.634-0500\t[################........]  nmdc-next.functional_annotation_agg  295MB/424MB  (69.6%)\n",
      "2023-03-01T14:05:21.634-0500\t[################........]  nmdc-next.functional_annotation_agg  298MB/424MB  (70.3%)\n",
      "2023-03-01T14:05:24.634-0500\t[#################.......]  nmdc-next.functional_annotation_agg  301MB/424MB  (70.9%)\n",
      "2023-03-01T14:05:27.634-0500\t[#################.......]  nmdc-next.functional_annotation_agg  303MB/424MB  (71.5%)\n",
      "2023-03-01T14:05:30.634-0500\t[#################.......]  nmdc-next.functional_annotation_agg  306MB/424MB  (72.1%)\n",
      "2023-03-01T14:05:33.635-0500\t[#################.......]  nmdc-next.functional_annotation_agg  308MB/424MB  (72.7%)\n",
      "2023-03-01T14:05:36.634-0500\t[#################.......]  nmdc-next.functional_annotation_agg  311MB/424MB  (73.3%)\n",
      "2023-03-01T14:05:39.635-0500\t[#################.......]  nmdc-next.functional_annotation_agg  313MB/424MB  (73.9%)\n",
      "2023-03-01T14:05:42.635-0500\t[#################.......]  nmdc-next.functional_annotation_agg  316MB/424MB  (74.5%)\n",
      "2023-03-01T14:05:45.635-0500\t[#################.......]  nmdc-next.functional_annotation_agg  318MB/424MB  (75.0%)\n",
      "2023-03-01T14:05:48.635-0500\t[##################......]  nmdc-next.functional_annotation_agg  321MB/424MB  (75.7%)\n",
      "2023-03-01T14:05:51.635-0500\t[##################......]  nmdc-next.functional_annotation_agg  324MB/424MB  (76.3%)\n",
      "2023-03-01T14:05:54.635-0500\t[##################......]  nmdc-next.functional_annotation_agg  326MB/424MB  (76.8%)\n",
      "2023-03-01T14:05:57.635-0500\t[##################......]  nmdc-next.functional_annotation_agg  329MB/424MB  (77.5%)\n",
      "2023-03-01T14:06:00.635-0500\t[##################......]  nmdc-next.functional_annotation_agg  330MB/424MB  (77.9%)\n",
      "2023-03-01T14:06:03.635-0500\t[##################......]  nmdc-next.functional_annotation_agg  333MB/424MB  (78.5%)\n",
      "2023-03-01T14:06:06.636-0500\t[##################......]  nmdc-next.functional_annotation_agg  335MB/424MB  (79.1%)\n",
      "2023-03-01T14:06:09.635-0500\t[###################.....]  nmdc-next.functional_annotation_agg  338MB/424MB  (79.7%)\n",
      "2023-03-01T14:06:12.635-0500\t[###################.....]  nmdc-next.functional_annotation_agg  341MB/424MB  (80.4%)\n",
      "2023-03-01T14:06:15.635-0500\t[###################.....]  nmdc-next.functional_annotation_agg  343MB/424MB  (80.9%)\n",
      "2023-03-01T14:06:18.635-0500\t[###################.....]  nmdc-next.functional_annotation_agg  346MB/424MB  (81.5%)\n",
      "2023-03-01T14:06:21.635-0500\t[###################.....]  nmdc-next.functional_annotation_agg  348MB/424MB  (82.1%)\n",
      "2023-03-01T14:06:24.636-0500\t[###################.....]  nmdc-next.functional_annotation_agg  350MB/424MB  (82.6%)\n",
      "2023-03-01T14:06:27.636-0500\t[###################.....]  nmdc-next.functional_annotation_agg  353MB/424MB  (83.3%)\n",
      "2023-03-01T14:06:30.636-0500\t[####################....]  nmdc-next.functional_annotation_agg  356MB/424MB  (83.9%)\n",
      "2023-03-01T14:06:33.636-0500\t[####################....]  nmdc-next.functional_annotation_agg  358MB/424MB  (84.4%)\n",
      "2023-03-01T14:06:36.636-0500\t[####################....]  nmdc-next.functional_annotation_agg  361MB/424MB  (85.1%)\n",
      "2023-03-01T14:06:39.636-0500\t[####################....]  nmdc-next.functional_annotation_agg  363MB/424MB  (85.7%)\n",
      "2023-03-01T14:06:42.636-0500\t[####################....]  nmdc-next.functional_annotation_agg  366MB/424MB  (86.2%)\n",
      "2023-03-01T14:06:45.636-0500\t[####################....]  nmdc-next.functional_annotation_agg  368MB/424MB  (86.9%)\n",
      "2023-03-01T14:06:48.636-0500\t[#####################...]  nmdc-next.functional_annotation_agg  371MB/424MB  (87.5%)\n",
      "2023-03-01T14:06:51.636-0500\t[#####################...]  nmdc-next.functional_annotation_agg  373MB/424MB  (88.0%)\n",
      "2023-03-01T14:06:54.637-0500\t[#####################...]  nmdc-next.functional_annotation_agg  376MB/424MB  (88.7%)\n",
      "2023-03-01T14:06:57.637-0500\t[#####################...]  nmdc-next.functional_annotation_agg  379MB/424MB  (89.3%)\n",
      "2023-03-01T14:07:00.643-0500\t[#####################...]  nmdc-next.functional_annotation_agg  381MB/424MB  (89.8%)\n",
      "2023-03-01T14:07:03.637-0500\t[#####################...]  nmdc-next.functional_annotation_agg  384MB/424MB  (90.5%)\n",
      "2023-03-01T14:07:06.637-0500\t[#####################...]  nmdc-next.functional_annotation_agg  386MB/424MB  (91.1%)\n",
      "2023-03-01T14:07:09.637-0500\t[#####################...]  nmdc-next.functional_annotation_agg  388MB/424MB  (91.6%)\n",
      "2023-03-01T14:07:12.637-0500\t[######################..]  nmdc-next.functional_annotation_agg  391MB/424MB  (92.2%)\n",
      "2023-03-01T14:07:15.638-0500\t[######################..]  nmdc-next.functional_annotation_agg  394MB/424MB  (92.9%)\n",
      "2023-03-01T14:07:18.637-0500\t[######################..]  nmdc-next.functional_annotation_agg  396MB/424MB  (93.4%)\n",
      "2023-03-01T14:07:21.638-0500\t[######################..]  nmdc-next.functional_annotation_agg  399MB/424MB  (94.1%)\n",
      "2023-03-01T14:07:24.638-0500\t[######################..]  nmdc-next.functional_annotation_agg  402MB/424MB  (94.8%)\n",
      "2023-03-01T14:07:27.637-0500\t[######################..]  nmdc-next.functional_annotation_agg  404MB/424MB  (95.2%)\n",
      "2023-03-01T14:07:30.638-0500\t[#######################.]  nmdc-next.functional_annotation_agg  406MB/424MB  (95.8%)\n",
      "2023-03-01T14:07:33.638-0500\t[#######################.]  nmdc-next.functional_annotation_agg  409MB/424MB  (96.4%)\n",
      "2023-03-01T14:07:36.638-0500\t[#######################.]  nmdc-next.functional_annotation_agg  411MB/424MB  (97.0%)\n",
      "2023-03-01T14:07:39.637-0500\t[#######################.]  nmdc-next.functional_annotation_agg  414MB/424MB  (97.6%)\n",
      "2023-03-01T14:07:42.638-0500\t[#######################.]  nmdc-next.functional_annotation_agg  417MB/424MB  (98.2%)\n",
      "2023-03-01T14:07:45.638-0500\t[#######################.]  nmdc-next.functional_annotation_agg  419MB/424MB  (98.7%)\n",
      "2023-03-01T14:07:48.638-0500\t[#######################.]  nmdc-next.functional_annotation_agg  421MB/424MB  (99.3%)\n",
      "2023-03-01T14:07:51.638-0500\t[########################]  nmdc-next.functional_annotation_agg  424MB/424MB  (100.0%)\n",
      "2023-03-01T14:07:51.689-0500\t[########################]  nmdc-next.functional_annotation_agg  424MB/424MB  (100.0%)\n",
      "2023-03-01T14:07:51.689-0500\tfinished restoring nmdc-next.functional_annotation_agg (3046840 documents, 0 failures)\n",
      "2023-03-01T14:07:51.689-0500\trestoring indexes for collection nmdc-next.study_set from metadata\n",
      "2023-03-01T14:07:51.689-0500\trestoring indexes for collection nmdc-next.data_object_set from metadata\n",
      "2023-03-01T14:07:51.689-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"data_object_type\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"data_object_type\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.689-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"file_size_bytes\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"file_size_bytes\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.689-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"url\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"url\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.689-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"md5_checksum\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"md5_checksum\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\trestoring indexes for collection nmdc-next.metabolomics_analysis_activity_set from metadata\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"has_credit_associations.applies_to_person.orcid\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"has_credit_associations.applies_to_person.orcid\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"has_credit_associations.applies_to_person.name\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"has_credit_associations.applies_to_person.name\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"has_credit_associations.applied_roles\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"has_credit_associations.applied_roles\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\trestoring indexes for collection nmdc-next.metaproteomics_analysis_activity_set from metadata\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:51.690-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.713-0500\trestoring indexes for collection nmdc-next.mags_activity_set from metadata\n",
      "2023-03-01T14:07:52.713-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.713-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.713-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\trestoring indexes for collection nmdc-next.metagenome_annotation_activity_set from metadata\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\trestoring indexes for collection nmdc-next.omics_processing_set from metadata\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"has_output\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"has_output\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"has_input\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"has_input\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"instrument_name\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"instrument_name\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.715-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"alternative_identifiers\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"alternative_identifiers\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.760-0500\trestoring indexes for collection nmdc-next.biosample_set from metadata\n",
      "2023-03-01T14:07:52.760-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ecosystem_type\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ecosystem_type\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.760-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ecosystem_subtype\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ecosystem_subtype\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.760-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"lat_lon.latitude\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"lat_lon.latitude\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.760-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"env_broad_scale.has_raw_value\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"env_broad_scale.has_raw_value\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.760-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"collection_date.has_raw_value\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"collection_date.has_raw_value\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.760-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"specific_ecosystem\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"specific_ecosystem\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"part_of_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"part_of\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ecosystem\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ecosystem\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"env_local_scale.has_raw_value\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"env_local_scale.has_raw_value\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"env_medium.has_raw_value\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"env_medium.has_raw_value\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"alternative_identifiers\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"alternative_identifiers\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ecosystem_category\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ecosystem_category\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"lat_lon.longitude\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"lat_lon.longitude\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:52.761-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:53.291-0500\trestoring indexes for collection nmdc-next.nom_analysis_activity_set from metadata\n",
      "2023-03-01T14:07:53.291-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:53.291-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:53.291-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:53.955-0500\trestoring indexes for collection nmdc-next.metagenome_assembly_set from metadata\n",
      "2023-03-01T14:07:53.955-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:53.955-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:53.955-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:53.955-0500\trestoring indexes for collection nmdc-next.functional_annotation_agg from metadata\n",
      "2023-03-01T14:07:53.955-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"metagenome_id2\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"metagenome_annotation_id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:54.092-0500\trestoring indexes for collection nmdc-next.metatranscriptome_activity_set from metadata\n",
      "2023-03-01T14:07:54.092-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:54.092-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:54.092-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:54.248-0500\trestoring indexes for collection nmdc-next.read_qc_analysis_activity_set from metadata\n",
      "2023-03-01T14:07:54.248-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"id_1\", \"unique\":true, \"v\":2}, Key:primitive.D{primitive.E{Key:\"id\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:54.248-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"name\":\"was_informed_by_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"was_informed_by\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:07:54.248-0500\tindex: &idx.IndexDocument{Options:primitive.M{\"background\":true, \"name\":\"ended_at_time_1\", \"v\":2}, Key:primitive.D{primitive.E{Key:\"ended_at_time\", Value:1}}, PartialFilterExpression:primitive.D(nil)}\n",
      "2023-03-01T14:08:00.039-0500\t3095234 document(s) restored successfully. 0 document(s) failed to restore.\n",
      "Took approx. 681.4387848377228 seconds\n"
     ]
    }
   ],
   "source": [
    "tic = time.time()\n",
    "\n",
    "!mongorestore --host {os.getenv(\"MONGO_HOST\")} \\\n",
    "     -u {os.getenv(\"MONGO_USERNAME\")} \\\n",
    "     -p {os.getenv(\"MONGO_PASSWORD\")} \\\n",
    "     --authenticationDatabase admin \\\n",
    "     -d {os.getenv(\"MONGO_DBNAME\") + \"-next\"} \\\n",
    "     --drop \\\n",
    "     {today_dir / os.getenv(\"MONGO_DBNAME\")}\n",
    "\n",
    "print(f\"Took approx. {time.time() - tic} seconds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92da8104-db07-4714-acb4-f81a0ade7d5d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (nmdc-runtime)",
   "language": "python",
   "name": "nmdc-runtime"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
